### PR TITLE
Add llvm-lit into nix devShell

### PR DESCRIPTION
**Pull request by Avimitin**
Tested with llvm lit built-in example tests directory.

### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [ ] Did you request a desired merge strategy?
- [ ] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- Internal or build-related (includes code refactoring/cleanup)

#### Desired Merge Strategy

- Rebase: You will rebase the PR onto master and it will be merged with a merge commit.

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.5.x`, `3.6.x`, or `5.x` depending on impact, API modification or big change: `6.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

**Comment by sequencer** [comment]
This is required by #3596 for my development environment. We don't have CI for it ;p

**Comment by dtzSiFive** [comment]
Maybe use nixpkgs-packaged lit? https://github.com/NixOS/nixpkgs/blob/75a52265bda7fd25e06e3a67dee3f0354e73243c/pkgs/development/python-modules/lit/default.nix .

`pythonPackages.lit` should work-- until channel gets that it'll be 15.0.6.

Anyway at very least can fetch this from official Pypi source instead of pulling out of the LLVM source tree which is a bit nicer :+1: .

File: nix/llvm-lit.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  llvm-lit = python3.pkgs.buildPythonPackage {
+    inherit src;
+
+    pname = "llvm-lit";
+    version = "17.0.3";
+    format = "setuptools";
+    sourceRoot = "source/llvm/utils/lit";
+    propagatedBuildInputs = [
+      python3.pkgs.setuptools
```
Comment (dtzSiFive) [inline nix/llvm-lit.nix:20]: I forget the rules for this for python (diff than normal packages :sweat: ) but offhand why propagate setuptools, for use of this as a build-dep of some other python thing?

**Comment by sequencer** [comment]
Yeah, I just found that, CC @Avimitin for contribution points :)

**Comment by Avimitin** [comment]
> Anyway at very least can fetch this from official Pypi source instead of pulling out of the LLVM source tree which is a bit nicer 👍 .

But finally we still need to build `FileCheck` and `not`, and maybe other binary dependencies. This derivation can provide a lit wrapper to handle this dependencies.

File: nix/llvm-lit.nix (hunk 1 above)
Comment (Avimitin) [reply nix/llvm-lit.nix:20]: Yes, this is my mistake. I should packaged it as an application not a library.

**Pull request merged**